pkg/controllers/jobset_controller_test.go

```diff
@@ -101,3 +102,164 @@ func TestIsJobFinished(t *testing.T) {
+								{
+									Key:      "job-name",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{jobName},
+								},
+							}},
+							TopologyKey: topologyKey,
+						},
+					},
+				},
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "job-name",
+									Operator: metav1.LabelSelectorOpNotIn,
+									Values:   []string{jobName, ""},
+								},
+							}},
+							TopologyKey: topologyKey,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "existing affinities should be appended to, not replaced",
+			job:  testJobWithAffinities(jobName),
+			wantAffinity: corev1.Affinity{
+				PodAffinity: &corev1.PodAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "label-foo",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{"value-foo"},
+								},
+							}},
+							TopologyKey: "topology.kubernetes.io/zone",
+						},
+						{
+							LabelSelector: &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "job-name",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{jobName},
+								},
+							}},
+							TopologyKey: topologyKey,
+						},
+					},
+				},
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+						{
+							LabelSelector: &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "label-bar",
+									Operator: metav1.LabelSelectorOpNotIn,
+									Values:   []string{"value-bar"},
+								},
+							}},
+							TopologyKey: "topology.kubernetes.io/zone",
+						},
+						{
+							LabelSelector: &metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "job-name",
+									Operator: metav1.LabelSelectorOpNotIn,
+									Values:   []string{jobName, ""},
+								},
+							}},
+							TopologyKey: topologyKey,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			setExclusiveAffinities(tc.job, topologyKey)
+			if diff := cmp.Diff(tc.wantAffinity.PodAffinity, tc.job.Spec.Template.Spec.Affinity.PodAffinity); diff != "" {
+				t.Errorf("unexpected diff in pod affinity (-want/+got): %s", diff)
+			}
+			if diff := cmp.Diff(tc.wantAffinity.PodAntiAffinity, tc.job.Spec.Template.Spec.Affinity.PodAntiAffinity); diff != "" {
+				t.Errorf("unexpected diff in pod anti-affinity (-want/+got): %s", diff)
+			}
+		})
+	}
+}
+
+func testJob(jobName string) *batchv1.Job {
```

Comment:
we have MakeJob in wrappers

Reply:
I thought about that as well but the MakeJob wrapper is for a batchv1.JobTemplateSpec, not a batchv1.Job, and I didn't think it was worth making a wrapper for this unit test - or do you think we should just to keep the pattern consistent throughout the codebase?

Reply:
I went ahead and updated the wrappers to have distinct `MakeJobTemplate` (for wrapping `batchv1.JobTemplateSpec`) and `MakeJob` for wrapping (`batchv1.Job)` and refactored the unit test to use the wrapper.